gensim/models/base_any2vec.py

```diff
@@ -1470,3 +1470,58 @@ def evaluate_word_pairs(self, pairs, delimiter='\t', restrict_vocab=300000,
 
         """
         return self.wv.evaluate_word_pairs(pairs, delimiter, restrict_vocab, case_insensitive, dummy4unknown)
+
+    def _find_similar_words(self, doc_vector=None, context_words_list=None, topn=None):
```

Comment:
These aren't necessarily "similar" words (as in the various other "similarity" methods), but *predicted* words, and the method name should reflect that. Separately, it's smewhat weird factoring to have a `doc_vector` input for a method in a Base class, and also weird to have two alternate parameters that are (not usually/sanely?) used together – that's a hint this *isn't* the truly shared forward-prop functionality – which would start with the shared path the method below does with `use_vector`.